Mathlib/Logic/Small/Basic.lean

```diff
@@ -23,7 +23,11 @@ section
 
 open Classical
 
-instance small_subtype (α : Type v) [Small.{w} α] (P : α → Prop) : Small.{w} { x // P x } :=
+-- TODO(timotree3): lower the priority on this instance?
+-- This instance applies to every synthesis problem of the form `Small ↥s` for some set `s`,
+-- but we have lots of instances of `Small` for specific set constructions.
+instance small_subtype (α : Type v) [Small.{w} α] (P : α → Prop) :
```

Comment:
I agree that this should have lower priority, but still higher than "generic" instances. Try `900`?

Reply:
Lowering the priority is causing compilation errors in later files that I don't know how to debug. Reverting for now


<details><summary>Example error</summary>

```text
[3216/4357]error: > LEAN_PATH=./.lake/packages/std/.lake/build/lib:./.lake/packages/Qq/.lake/build/lib:./.lake/packages/aesop/.lake/build/lib:./.lake/packages/proofwidgets/.lake/build/lib:./.lake/packages/Cli/.lake/build/lib:./.lake/packages/importGraph/.lake/build/lib:./.lake/build/lib LD_LIBRARY_PATH=./.lake/build/lib /home/lean/.elan/toolchains/leanprover--lean4---v4.7.0-rc2/bin/lean -Dpp.unicode.fun=true -Dpp.proofs.withType=false -DautoImplicit=false -DrelaxedAutoImplicit=false ./././Mathlib/Algebra/Module/Injective.lean -R ././. -o ./.lake/build/lib/Mathlib/Algebra/Module/Injective.olean -i ./.lake/build/lib/Mathlib/Algebra/Module/Injective.ilean -c ./.lake/build/ir/Mathlib/Algebra/Module/Injective.c
error: stdout:
Error: ./././Mathlib/Algebra/Module/Injective.lean:468:9: error: stuck at solving universe constraint
  v =?= max u ?u.297867
while trying to unify
  Shrink.{max u ?u.297867, u} ↥I →ₗ[R] ↥I : Type (max v u)
with
  LinearMap.{u, max u ?u.297867, u, u} (RingHom.id R) (Shrink.{max u ?u.297867, u} ↥I)
    ↥I : Type (max (max u ?u.297867) u)
Error: ./././Mathlib/Algebra/Module/Injective.lean:464:0: error: stuck at solving universe constraint
  v =?= max u ?u.297867
while trying to unify
  Shrink.{max u ?u.297867, u} ↥I →ₗ[R] ↥I : Type (max v u)
with
  LinearMap.{u, max u ?u.297867, u, u} (RingHom.id R) (Shrink.{max u ?u.297867, u} ↥I)
    ↥I : Type (max (max u ?u.297867) u)
error: external command `/home/lean/.elan/toolchains/leanprover--lean4---v4.7.0-rc2/bin/lean` exited with code 1
```

</details>